src/app/scripts/cac/control/cac-control-directions.js

```diff
@@ -164,7 +180,15 @@ CAC.Control.Directions = (function (_, $, moment, Control, Routing, UserPreferen
             itineraryListControl.show();
             // highlight first itinerary in sidebar as well as on map
             findItineraryBlock(currentItinerary.id).addClass(options.selectors.selectedItineraryClass);
+            planTripRequest = null;
```

Comment:
If `planTrip` got invoked again while the previous one was in this block, it would call `planTripRequest.reject()` after it had already resolved.  Maybe that's not possible in practice, and maybe it wouldn't be a big deal if it happened, but I'd be inclined to put this at the top of the callback just in case, since I don't see any disadvantage to doing so.

Reply:
Reasoning in nullifying the request as late as possible is so that display of known outdated results can be aborted, rather than doing unnecessary work and potentially delaying display of the newer response. According to this [docs example](https://api.jquery.com/deferred.promise/), though, it looks like it would have no effect on a completed promise.

Reply:
Ok, yeah, no difference then, and calling `reject` after it's `resolved` is ignored (I was thinking it might throw an error).  So this is fine as-is.